diff --git a/internal/common/module_mappings.bzl b/internal/common/module_mappings.bzl
index c523b50a..0ca35995 100644
--- a/internal/common/module_mappings.bzl
+++ b/internal/common/module_mappings.bzl
@@ -80,7 +80,7 @@ def get_module_mappings(label, attrs, srcs = [], workspace_name = None, mappings
             mr = "%s/%s" % (workspace_name, mr)
         elif label.workspace_root:
             mr = "%s/%s" % (label.workspace_root, mr)
-        if attrs.module_root and attrs.module_root != ".":
+        if hasattr(attrs, "module_root") and attrs.module_root and attrs.module_root != ".":
             if attrs.module_root.endswith(".ts"):
                 if workspace_name:
                     # workspace_name is set only when doing module mapping for runtime.
diff --git a/internal/node/bazel_require_script.js b/internal/node/bazel_require_script.js
index 3d062b29..1bf613e3 100644
--- a/internal/node/bazel_require_script.js
+++ b/internal/node/bazel_require_script.js
@@ -501,7 +501,7 @@ exports.escapeFunction = (root) => {
         return false;
     }
     function isOutPath(str) {
-        return !root || (!str.startsWith(root + path.sep) && str !== root);
+        return !root || (!str.startsWith(root + path.sep) && str !== root) || str.startsWith(root + path.sep + 'npm' + path.sep);
     }
     return { isEscape, isOutPath };
 };
diff --git a/internal/node/node.bzl b/internal/node/node.bzl
index 2c0bfefd..2043213c 100644
--- a/internal/node/node.bzl
+++ b/internal/node/node.bzl
@@ -25,7 +25,7 @@ load("//internal/common:expand_into_runfiles.bzl", "expand_location_into_runfile
 load("//internal/common:module_mappings.bzl", "module_mappings_runtime_aspect")
 load("//internal/common:path_utils.bzl", "strip_external")
 load("//internal/common:windows_utils.bzl", "create_windows_native_launcher_script", "is_windows")
-load("//internal/linker:link_node_modules.bzl", "write_node_modules_manifest")
+load("//internal/linker:link_node_modules.bzl", "write_node_modules_manifest", "module_mappings_aspect")
 load("//internal/node:node_repositories.bzl", "BUILT_IN_NODE_PLATFORMS")
 
 def _trim_package_node_modules(package_name):
@@ -277,7 +277,7 @@ _NODEJS_EXECUTABLE_ATTRS = {
     "data": attr.label_list(
         doc = """Runtime dependencies which may be loaded during execution.""",
         allow_files = True,
-        aspects = [node_modules_aspect, module_mappings_runtime_aspect],
+        aspects = [node_modules_aspect, module_mappings_aspect, module_mappings_runtime_aspect],
     ),
     "default_env_vars": attr.string_list(
         doc = """Default environment variables that are added to `configuration_env_vars`.
